File: Array/TypeScript/_1480_RunningSumof1dArray.ts
/**
 * Iteration
 * Time Complexity: BigO(N)
 * Space Complexity: BigO(N)
 */
function runningSum(nums: number[]): number[] {
  let sum = 0;
  return nums.map((num) => (sum += num));
}

console.log(runningSum([1, 2, 3, 4]));
console.log(runningSum([1, 1, 1, 1, 1]));
console.log(runningSum([3, 1, 2, 10, 1]));

File: Array/Python/_1480_RunningSumof1dArray.py
from typing import List


class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        result = []
        sum = 0
        for num in nums:
            sum += num
            result.append(sum)
        return result


ins = Solution

print(ins.runningSum(ins, [1, 2, 3, 4]))
print(ins.runningSum(ins, [1, 1, 1, 1, 1]))
print(ins.runningSum(ins, [3, 1, 2, 10, 1]))
